crypto/multisig/compact_bit_array.go

```diff
@@ -71,6 +71,21 @@ func (bA *CompactBitArray) SetIndex(i int, v bool) bool {
 	return true
 }
 
+// NumOfTrueBitsBefore returns the location of the given index, among the
+// values in the bit array that are set to true.
+// e.g. if bA = _XX_X_X, NumOfTrueBitsBefore(4) = 2, since
```

Comment:
"since ... value at index 4 .... is true" 
"since" seems like a strange modifier and the following statement a bit orthogonal.

Reply:
Maybe it make sense to let the user specify 0 to count the total number of true bits?
And then we can comment that, which makes it clear that you don't want to enter NumTrueBitsBefore(0) when you really mean NumTrueBitsBefore(1).

Reply:
I feel like making edge cases like letting 0 counting the number of true bits makes it more confusing. I do definitely agree with your point about the comments here!